library/std/intrinsic.qs

```diff
@@ -226,10 +226,42 @@ namespace Microsoft.Quantum.Intrinsic {
     /// ```qsharp
     /// Measure([PauliZ], [qubit]);
     /// ```
+    @Config(Full)
     operation M(qubit : Qubit) : Result {
         __quantum__qis__m__body(qubit)
     }
 
+    /// # Summary
+    /// Performs a measurement of a single qubit in the
+    /// Pauli _Z_ basis.
+    ///
+    /// # Input
+    /// ## qubit
+    /// Qubit to be measured.
+    ///
+    /// # Output
+    /// `Zero` if the +1 eigenvalue is observed, and `One` if
+    /// the -1 eigenvalue is observed.
+    ///
+    /// # Remarks
+    /// The output result is given by
+    /// the distribution
+    /// $$
+    /// \begin{align}
+    ///     \Pr(\texttt{Zero} | \ket{\psi}) =
+    ///         \braket{\psi | 0} \braket{0 | \psi}.
+    /// \end{align}
+    /// $$
+    ///
+    /// Equivalent to:
+    /// ```qsharp
+    /// Measure([PauliZ], [qubit]);
+    /// ```
+    @Config(Base)
+    operation M(qubit : Qubit) : Result {
```

Comment:
Why is a different version of `M` needed for base profile? How does it help to defer measurements?

Reply:
We had the chance to discuss this, but capturing the output here. The idea here is that because measurement needs to leave the qubit in the measured state, the only way to do that while remaining base profile compliant (defering measurements and avoiding qubit reuse) is to use an additional qubit and entanglement to perform the measurement. This lets us ensure that the program will still be correct when measurements are "pulled down" to the end of the program. We should ensure that as we extend the values allowed by `@Config` we don't force ourselves to duplicate an identical function definitions, which implies some version of additive `@Config` options.